tests/saveToCase.js

```diff
@@ -100,6 +100,9 @@ define([
             assert.equal(index.p.user_id, "/data/meta/userID");
             assert.equal(index.p.case_id, "/data/meta/caseID");
             util.assertXmlEqual(call("createXML"), INDEX_PROPERTY_XML);
+
+            // clean up from clickQuestion
+            window.history.replaceState(null, null, '/');
```

Comment:
These were the only ones that were causing other tests to fail, but you are right that it is probably better to clean up from all `clickQuestion`s in the other places.

I tried looking for something like `addCleanup`, so I could clean up automatically but it seems like it doesn't exist in mocha (https://github.com/mochajs/mocha/issues/2741). 

The other option I could see is to run this in a global `afterEach` which would run after every single test, but that seemed like a bit of overkill to me.

Reply:
Could you add this at the end of `clickQuestion()`? It's not clear to me why this is even necessary.

Reply:
If you call `clickQuestion` it adds question path to the URL, then in subsequent tests, when the form loads, it immediately selects the question that has that path. For things like save-to-case this actually changes the behaviour, because when you click the question it adds new empty values for things like the update properties. 

I suppose I could add it to the end of `clickQuestion`, although that would lead to different behaviour than what happens "in the wild" (which seems like what some of these test utils do anyway, so it might not be surprising).

Reply:
Another option would be to clear the hashtag path in `util.loadXML()` before a new form is loaded. It may be necessary to manually clean up in some tests that do not use `util.loadXML()`. That would be a little dirty because the test is cleaning up state created by the test that came before it. However, I consider having to remember to add this cleanup to every test that calls `clickQuestion()` a little dirty as well because it's not obvious and easy to forget.